da/avail/datasubmit/submitdata.go

```diff
@@ -0,0 +1,88 @@
+package datasubmit
+
+import (
+	"errors"
+
+	gsrpc "github.com/centrifuge/go-substrate-rpc-client/v4"
+	"github.com/centrifuge/go-substrate-rpc-client/v4/signature"
+	"github.com/centrifuge/go-substrate-rpc-client/v4/types"
+)
+
+// submit data submits the extrinsic through substrate api
+func SubmitData(apiURL string, seed string, appID int, data []byte) error {
+
+	// if app id is greater than 0 then it must be created before submitting data
+	if appID == 0 {
+		return errors.New("AppID cant be 0")
+	}
+
+	api, err := gsrpc.NewSubstrateAPI(apiURL)
+	if err != nil {
+		return err
+	}
+
+	meta, err := api.RPC.State.GetMetadataLatest()
+	if err != nil {
+		return err
+	}
+
+	c, err := types.NewCall(meta, "DataAvailability.submit_data", data)
+	if err != nil {
+		return err
+	}
+
+	// Create the extrinsic
+	ext := types.NewExtrinsic(c)
+
+	genesisHash, err := api.RPC.Chain.GetBlockHash(0)
+	if err != nil {
+		return err
+	}
+
+	rv, err := api.RPC.State.GetRuntimeVersionLatest()
+	if err != nil {
+		return err
+	}
+
+	keyringPair, err := signature.KeyringPairFromSecret(seed, 42)
```

Comment:
Not sure if this is a secure way of doing it. Did you check other cosmos implementations on how this is being handled?